nativelib/src/main/resources/gc/immix/LargeAllocator.h

```diff
@@ -0,0 +1,39 @@
+#ifndef IMMIX_LARGEALLOCATOR_H
+#define IMMIX_LARGEALLOCATOR_H
+
+#include "datastructures/Bitmap.h"
+#include "GCTypes.h"
+#include "Constants.h"
+#include "headers/ObjectHeader.h"
+
+#define FREE_LIST_COUNT                                                        \
+    (LARGE_OBJECT_MAX_SIZE_BITS - LARGE_OBJECT_MIN_SIZE_BITS + 1)
+
+typedef struct Chunk Chunk;
+
+struct Chunk {
+    ObjectHeader header;
+    Chunk *next;
+};
+
+typedef struct {
+    Chunk *first;
+    Chunk *last;
+} FreeList;
```

Comment:
Free lists have operations defined on them, sounds like a thing that one would put into datastructures.